File: sanghyun/programmers-2/올바른 괄호.java
class Solution {
    boolean solution(String s) {
        boolean answer = true;
        int cnt = 0;
        for (int i = 0; i < s.length(); i++) {
            char c = s.charAt(i);
            if(c == '('){
                cnt++;
            }else {
                cnt--;
            }
            if(cnt < 0){
                return false;
            }
        }
        if(cnt == 0){
            return true;
        }else{
            return false;
        }
    }
}

File: sanghoon/programmers-2/올바른 괄호.py
def solution(s):
    cnt = 0
    for i in s:
        if i == "(":
            cnt += 1
        elif i == ")":
            cnt -= 1

        if cnt < 0:
            return False
            break

    if cnt == 0:
        return True
    else:
        return False
s = "(()("
solution(s)
